core/internal/filetransfer/file_transfer_manager.go

```diff
@@ -75,69 +74,34 @@ func NewFileTransferManager(opts ...FileTransferManagerOption) FileTransferManag
 	return &fm
 }
 
-func (fm *fileTransferManager) AddTask(task *Task) {
-	fm.logger.Debug("fileTransfer: adding upload task", "path", task.Path, "url", task.Url)
+func (fm *fileTransferManager) AddTask(task Task) {
+	fm.logger.Debug("fileTransferManager: AddTask: adding task", "task", task.String())
 
 	fm.wg.Add(1)
 	go func() {
 		defer fm.wg.Done()
 
 		// Guard by a semaphore to limit number of concurrent uploads.
 		fm.semaphore <- struct{}{}
-		task.Err = fm.transfer(task)
+		err := fm.transfer(task)
+		task.SetError(err)
```

Comment:
directly assigning should be fine... if it only updates the Err field

Reply:
`filetransfer.Task` is an interface so we can't set the field directly here unfortunately